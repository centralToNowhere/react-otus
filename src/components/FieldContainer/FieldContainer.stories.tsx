import React from "react";
import { Story, Meta } from "@storybook/react";
import { rest } from "msw";
import {
  FieldContainer,
  FieldContainerProps,
} from "@/components/FieldContainer/FieldContainer";
import figures from "@/components/Field/figures.json";

export default {
  title: "Components/FieldContainer",
  component: FieldContainer,
  args: {
    cellSize: 40,
    cellsBetweenChars: 1,
    fieldWidthPx: 600,
    fieldHeightPx: 400,
  },
  parameters: {
    msw: [
      rest.get("/figures.json", (req, res, ctx) => {
        return res(ctx.json(figures));
      }),
    ],
  },
} as Meta;

const Template: Story<FieldContainerProps> = (args) => {
  return <FieldContainer {...args} />;
};

export const FieldContainerSmall = Template.bind({});
FieldContainerSmall.args = {
  cellSize: 20,
  fieldWidthPx: 600,
  fieldHeightPx: 300,
};

export const FieldContainerBig = Template.bind({});
FieldContainerBig.args = {
  cellSize: 10,
  fieldWidthPx: 800,
  fieldHeightPx: 300,
};
