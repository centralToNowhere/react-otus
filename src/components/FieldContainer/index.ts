export { FieldContainer } from "./FieldContainer";
export type { FieldContainerProps } from "./FieldContainer";
