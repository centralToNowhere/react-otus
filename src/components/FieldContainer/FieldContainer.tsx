import React from "react";
import styled from "@emotion/styled";
import { Form, isPossibleInput } from "@/components/Form";
import { Field, IFiguresMap } from "@/components/Field";

export type FieldContainerProps = typeof FieldContainer.defaultProps & {
  cellSize?: number;
  cellsBetweenChars?: number;
  fieldWidthPx?: number;
  fieldHeightPx?: number;
};

interface IFieldContainerState {
  inputString: string;
  figuresMapLoaded: boolean;
  figuresMap: IFiguresMap["figures"];
}

export class FieldContainer extends React.Component<
  FieldContainerProps,
  IFieldContainerState
> {
  static defaultProps = {
    cellSize: 30,
    cellsBetweenChars: 1,
    fieldWidthPx: 800,
    fieldHeightPx: 400,
  };

  constructor(props: FieldContainerProps) {
    super(props);

    this.onInputChange = this.onInputChange.bind(this);

    this.state = {
      inputString: "",
      figuresMapLoaded: false,
      figuresMap: {},
    };
  }

  onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    if (isPossibleInput(e.target.value)) {
      this.setState({
        inputString: String(e.target.value),
      });
    }
  };

  fetchFiguresData(): Promise<IFiguresMap> {
    return fetch("/figures.json", {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
      },
    })
      .then((response) => response.json())
      .then((data: IFiguresMap) => {
        return data;
      });
  }

  componentDidMount() {
    this.fetchFiguresData().then((data: IFiguresMap) => {
      this.setState((prevState) => {
        return {
          ...prevState,
          figuresMapLoaded: true,
          figuresMap: data.figures,
        };
      });
    });
  }

  render() {
    return (
      <Container>
        {this.state.figuresMapLoaded && (
          <>
            <Field
              cellSize={this.props.cellSize}
              cellsBetweenChars={this.props.cellsBetweenChars}
              inputString={this.state.inputString}
              figuresMap={this.state.figuresMap}
              fieldWidthPx={this.props.fieldWidthPx}
              fieldHeightPX={this.props.fieldHeightPx}
            />
            <Form
              inputString={this.state.inputString}
              onChange={this.onInputChange}
            />
          </>
        )}
      </Container>
    );
  }
}

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;
