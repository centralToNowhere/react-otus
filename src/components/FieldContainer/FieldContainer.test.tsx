import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { FieldContainer } from "@/components/FieldContainer/FieldContainer";
import figures from "@/components/Field/figures.json";

const inputLabel = "Введите целое число:";

beforeAll(() => {
  fetchMock.doMock();
  fetchMock.mockIf("/figures.json", JSON.stringify(figures));
});

describe("FieldContainer tests", () => {
  it("should correctly render field and form", async () => {
    const { asFragment } = render(<FieldContainer />);

    await waitFor(() => {
      const field = screen.getByTestId("field");
      expect(field).toBeInTheDocument();
    });

    await waitFor(() => {
      const form = screen.getByTestId("interactive-number-drawing-form");
      expect(form).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it("should call send request", () => {
    fetch("/figures.json").then((response) => response.json());

    expect(fetchMock).toBeCalledTimes(1);
  });

  it("should print allowable input characters", async () => {
    render(<FieldContainer />);

    await screen.findByLabelText(inputLabel);

    const input: HTMLInputElement = screen.getByLabelText(inputLabel);
    userEvent.type(input, "1234{enter}");

    expect(input).toHaveValue("1234");
  });

  it("should cancel printing all other characters than [0-9]", async () => {
    render(<FieldContainer />);

    await screen.findByLabelText(inputLabel);

    const input: HTMLInputElement = screen.getByLabelText(inputLabel);
    userEvent.type(input, "!1$RE%3$@6#$");

    expect(input).toHaveValue("136");
  });
});
