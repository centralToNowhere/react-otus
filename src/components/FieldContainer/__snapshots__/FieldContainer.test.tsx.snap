// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldContainer tests should correctly render field and form 1`] = `
<DocumentFragment>
  <div
    class="css-1oskuwq"
  >
    <div
      class="css-1ru2519"
      data-testid="field"
    >
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
      <div
        class="css-11pa12e"
        data-testid="cell"
        size="30"
      />
    </div>
    <form
      class="css-p3c1cq"
      data-testid="interactive-number-drawing-form"
    >
      <label
        class="css-0"
        for="number-input"
      >
        Введите целое число:
      </label>
      <input
        class="css-0"
        id="number-input"
        name="number-input"
        type="text"
        value=""
      />
    </form>
  </div>
</DocumentFragment>
`;
